suricata/update/main.py

```diff
@@ -199,21 +199,29 @@ def fetch(self, url):
         logger.info("Done.")
         return self.extract_files(tmp_filename)
 
-    def run(self, url=None, files=None):
+    def run(self, url, files=None):
         if files is None:
             files = {}
         if url:
             try:
                 fetched = self.fetch(url)
-                files.update(fetched)
+                for key in fetched:
+                    safe_filename = self.safe_filename(url[0], key)
+                    files[safe_filename] = fetched[key]
             except URLError as err:
                 url = url[0] if isinstance(url, tuple) else url
                 logger.error("Failed to fetch %s: %s", url, err)
-        else:
-            for url in self.args.url:
-                files.update(self.fetch(url))
         return files
 
+    def safe_filename(self, url, filename):
+        """Create a safe/conflict free filename from the URL and the full path
+        of a file inside the archive.
+        """
+        scheme_end = url.find("://")
+        if scheme_end > 0:
+            return "{}!{}".format(url[scheme_end + 1:], filename)
+        return "{}!{}".format(url, filename)
+
```

Comment:
This works well. I have a few questions though. I see debug info like:
```
1/4/2020 -- 17:03:00 - <Debug> -- Parsing //rules.emergingthreats.net/open/suricata-5.0.1/emerging.rules.tar.gz!rules/emerging-web_client.rules.
```
which is not clickable any more but do we want the front `//` in the file name?

Also, I see that in the end with `--no-merge` option, some files written in `/var/lib/suricata/rules` are named `trafficid.rules!trafficid.rules`, do we want this nomenclature?

Reply:
To me I see '//' as this came from something over a URL.. But could remove it.

Have the leading text including the '!' should never be in the output. I'll have to fix that. Thanks for catching it.